### PR TITLE
fix: only show public newsletters on CPT archive pages

### LINKED ISSUE #403: Newsletter CPT archive shows non-public posts
Example: https://montanafreepress.org/newsletter/

Expected: only newsletter posts marked as "public" should be shown here.

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

#398 introduced a regression on Newsletter CPT archives where all published/sent newsletters are shown, whether or not they are marked as public. This fixes the query on those archives so that only those newsletters marked as publicly visible will appear, as in category and tag archives.

Closes #403.

### How to test the changes in this Pull Request:

1. Create and send several newsletters, but mark only some of them as publicly visible.
2. On `master`, visit the `https://<your_site_url>/newsletter` archive page (or whatever the slug you've defined for newsletters in Newsletter settings). Observe that all sent newsletters are displayed here, even those not marked as publicly visible.
3. Check out this branch and refresh the archive page. Confirm that now only those newsletters marked as publicly visible are shown.
4. Change the "publicly visible" setting for some newsletters and confirm that the posts shown in the archive are updated accordingly.
5. Apply some tags and categories to your newsletters, both public and not. Visit the archive pages for those terms and confirm that you still see all published posts with the term, plus only the newsletters that are marked as publicly visible. (This behavior should be unchanged from `master`).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by adekbadek** [comment]
@claudiulodro Could you please take a look? I remember you had some remarks about potential issues with query filtering.

**Review by adekbadek — COMMENTED** [review]
I did have the public newsletters archive page slug set (in case that matters), but on `master` I did _not_ see the non-public newsletters. 

What's more, and probably unrelated, after unsetting the `newspack_newsletters_public_posts_slug` option the page at `/newsletters` was not found. 

But perhaps I'm missing a step here? I did `wp cache flush`, but it did not fix the issues.

**Comment by dkoo** [comment]
> I did have the public newsletters archive page slug set (in case that matters), but on master I did not see the non-public newsletters.

Hmm, you're right—it's not _all_ published-but-private newsletters that show on `master`. I was able to reproduce the issue with the `master` code on the Model Home test site, if that helps. After a bit of investigation it's possible that only newsletters that were published prior to #398 are affected by this, but I'm not sure why that would be.

> What's more, and probably unrelated, after unsetting the `newspack_newsletters_public_posts_slug` option the page at /newsletters was not found.

This sounds like the expected behavior, no? That option sets the URL slug for accessing the CPT archive page for newsletters, so if you unset it, it makes sense that the archive wouldn't be accessible, correct?

**Comment by adekbadek** [comment]
> This sounds like the expected behavior, no? That option sets the URL slug for accessing the CPT archive page for newsletters, so if you unset it, it makes sense that the archive wouldn't be accessible, correct?

There's a [default value here](https://github.com/Automattic/newspack-newsletters/blob/381dc5174806118d50da07d7bb89d40f073146a1/includes/class-newspack-newsletters.php#L422), so after unsetting the option I'd expect to see the newsletters at `/newsletter`

**Comment by dkoo** [comment]
> There's a [default value here](https://github.com/Automattic/newspack-newsletters/blob/381dc5174806118d50da07d7bb89d40f073146a1/includes/class-newspack-newsletters.php#L422), so after unsetting the option I'd expect to see the newsletters at `/newsletter` 

Ah, you're right. I updated the behavior in a96d9b0 so that if the option is updated to an empty value, it resets it to the default. This way the archive is always available at the `newsletter` slug by default.

File: includes/class-newspack-newsletters-settings.php [hunk 1]
```diff
@@ -196,8 +196,9 @@ public static function page_init() {
 	public static function newspack_newsletters_settings_callback( $setting ) {
 		$key         = $setting['key'];
 		$type        = $setting['type'];
+		$default     = $setting['default'];
```
[the remark this answers is not in the document]
Comment (dkoo) [reply includes/class-newspack-newsletters-settings.php]: Ah yes, I had fixed that locally in a58c5a0 but forgot to push it.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**